src/ServiceControl/ExternalIntegrations/EventDispatcher.cs

```diff
@@ -143,8 +145,15 @@ private bool TryDispatchEventBatch()
                 {
                     Logger.Debug($"Dispatching {allContexts.Length} events.");
                 }
-                var eventsToBePublished = eventPublishers.SelectMany(p => p.PublishEventsForOwnContexts(allContexts, session));
 
+                var eventsToBePublished = new List<object>();
+                foreach (var publisher in eventPublishers)
```

Comment:
Maybe await the entire set of tasks instead of each one individually? 

```CSharp
var eventsToBePublished = Task.WaitAll(eventPublishers.Select(publisher  => publisher.PublishEventsForOwnContexts(allContexts, session).ConfigureAwait(false)).SelectMany();
```

Or something, I'd have to check the syntax properly

Reply:
See https://github.com/Particular/ServiceControl/pull/1176#discussion_r194648515